### ISSUE TITLE
[terra-form-select] Screen Reader is not reading the index value for options

### ISSUE THREAD

**Issue by saket2403** [cerner/terra-core#3325]
# Bug Report

## Description
Voice Over starts reading by announcing select has 5 options and then reads out the options as we tab through them without announcing "1 off 5", "2 off 5", etc along with the name.

## Steps to Reproduce

1. Open any example on [dev-site](https://engineering.cerner.com/terra-core/components/terra-form-select/form-select/combobox).
2. Navigate to the select 
3. Turn on VO and tab through the options.

## Expected Behavior
It should read index values along with the option name.

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

## @ Mentions
@ryanthemanuel , @neilpfeiffer

**Comment by neilpfeiffer** [comment]
May need to test the combination of aria roles along with this, but should be as simple as adding two additional aria attributes:
- [aria-posinset](https://www.w3.org/TR/wai-aria-1.1/#aria-posinset)
- [aria-setsize](https://www.w3.org/TR/wai-aria-1.1/#aria-setsize)

Reference: [Mozilla › developer: WAI-ARIA basics](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/WAI-ARIA_basics#:~:text=aria%2Dsetsize%20%E2%80%94%20This%20property%20allows,series%20an%20element%20is%20in.)

If this doesn't work, we may need to construct a similar screenreader phrase manually for each item using an aria-label, labelled-by, described-by, etc.

**Comment by saket2403** [comment]
While testing with VO, I found out that the VO reads the index value if the focus is on the option element. But it reads only for the first option and then the select closes on using the VO navigation keys (CapsLock+Right Arrow) and we are unable to navigate to the next option. On using the Up and down arrow keys the VO focus is not inside the select therefore it reads only the option value and no aria-attributes and roles, etc. Should I go ahead and further investigate into moving the focus inside the select and prevent it from closing on using VO navigation keys, or will there be any accessibility issues in doing so.
cc @neilpfeiffer

**Comment by supreethmr** [comment]
I think ARIA attributes `aria-posinset` and `aria-setsize` will work only when the list item has focus on it as you mentioned. So this would be not ideal for form-select since items in form select are not accessible through Tab keys / gets focus on navigate unlike terra-menu. 

Tweaking the existing focus behavior might break accessibility of the component so I think handling this through with custom-text ([adding text for screen readers to read out initial state of form-select and selected state of item](https://github.com/cerner/terra-core/blob/7cf568bd47996ec173dfed75c22746d4891ade26/packages/terra-form-select/src/single/Frame.jsx#L384-L418)) likely to be having low-impact on existing behavior / functionality of form-select. 

I think we might start looking into updating [this visuallyHiddenComponent (this. visuallyHiddenComponent.innerText)](https://github.com/cerner/terra-core/blob/main/packages/terra-form-select/src/single/Menu.jsx#L312)on keydown event to update the index of active item for screen-readers.

**Comment by saket2403** [comment]
I have a question regarding opt-groups. Lets say there are 5 options in one group and 4 in another so a total of 9 options. Should the VO read the option in Group 1 as `1 of 5` **OR** `1 of 9` and similarly in Group 2 as `1 of 4` **OR** `6 of 9`. Currently I have coded it to read index for each group separately ie: `1 of 5`  and `1 of 4` for another. cc @neilpfeiffer .

**Comment by saket2403** [comment]
I have tried a couple of approaches to implement index for native-select:

- Used `aria-setsize` and `aria-posinset` on `option` element, but it is not getting read out by VO as active element is not the option elemnt when navigating using VO.
- An alternative was visually hidden component similar to what I have used for all other variants, but it is not working option element as the child element cannot be any other element apart from number or text.

I used the 2nd approach in all other variants and they are working fine ([Branch Link](https://github.com/cerner/terra-core/tree/select-setsize)).
Is there any alternative that can be used for Native select? cc @dkasper-was-taken .

**Comment by scottwilmarth** [comment]
If terra-form-select is the native HTML select component, Cerner should not be customizing the experience of it. We would not want to custom code a fix for a native component. The native component is not our liability. The issue should be logged with Apple VoiceOver for Apple to address VoiceOver shortcomings. This prevents any future VoiceOver updates from breaking our code which would make it our liability. @neilpfeiffer @mjhenkes

**Comment by StephenEsser** [comment]
How are we determining the "correct" accessibility behavior of the dropdown? How did we make the decision the select should read the `x of y` options? Is there an w3c/aria or similar web standards document available with requirements?

Regarding the comments about the native select. The native select does not announce the index and in practice I use the native select as a guide for behavior. The native select is not always perfect and may not be a good role model here, but if we are pursuing these changes I'd like to see or discuss the rationale behind our decision to better understand our intentions. 

Doing some browsing I occasionally stumble into this [custom select implementation from deque](https://pattern-library.dequelabs.com/components/selects). Also listed [here](https://www.webaxe.org/accessible-custom-select-dropdowns/). I think these are good examples to compare against. I noticed their announcements seem to work as outline in this issue. When I open our own implementation in Google Chrome using voiceover I can see it is trying to read the announcement `x of y`, but is immediately switched to only read the display. I think this is because of aria live regions overwriting the VO active-descendent/list announcements.

**Comment by saket2403** [comment]
We are waiting for a confirmation from UX and A11y teams on the `x of y` phrase read out for index by VO, will be kept/changed accordingly.
Native Select will be left as is , as discussed with scott above, index will be implemented for all other variants.
I discussed the immediate switching behavior with @supreethmr, and it looks like removing these lines of [code](https://github.com/cerner/terra-core/blob/4030f4e74a553299e30b05c56ec5dca97f30fc54/packages/terra-form-select/src/single/Frame.jsx#L522-L526) and increasing the speech rate of voice over allows the Vo to read the index as well.
